runtime/plaid/src/loader/mod.rs

```diff
@@ -428,6 +500,68 @@ pub async fn load(
     Ok(modules)
 }
 
+/// Verifies a signature file against a module's hash and ensures it is signed by an authorized signer.
+fn verify_signature_file(
+    signature_path: DirEntry,
+    module_hash: &[u8],
+    signing: &ModuleSigningConfiguration,
+    module_name: &str,
+) -> Result<(), Errors> {
+    // Convert file name to string for logging and filtering.
+    let sig_filename = signature_path.file_name().to_string_lossy().to_string();
+    if !sig_filename.ends_with(".sig") {
+        return Err(Errors::BadFilename);
+    }
+
+    // Try to read the signature file.
+    let signature_content = match std::fs::read_to_string(signature_path.path()) {
+        Ok(content) => content,
+        Err(e) => {
+            error!(
+                "Failed to read signature at [{:?}]. Error: {}",
+                signature_path, e
+            );
+            return Err(Errors::MissingSignatureFile);
+        }
+    };
+
+    // Parse and validate the signature.
+    let ssh_signature = match SshSignature::from_armored_string(&signature_content) {
+        Ok(sig) => sig,
+        Err(e) => {
+            error!("Invalid signature provided for {}", module_name);
+            return Err(Errors::SshCertsError(e));
+        }
+    };
+
+    let pubkey = ssh_signature.pubkey.clone();
+
+    // Check if the signature was made by an authorized signer.
+    if !signing
+        .authorized_signers
+        .iter()
+        .any(|s| *s == pubkey.fingerprint().to_string())
+    {
+        error!(
+            "{} was signed by an unexpected signer: {}",
+            module_name,
+            pubkey.fingerprint()
+        );
+        return Err(Errors::UnauthorizedSigner);
+    }
+
+    // Verify the signature using the module hash.
+    VerifiedSshSignature::from_ssh_signature(
+        module_hash,
+        ssh_signature,
+        &signing.signature_namespace,
+        Some(pubkey),
+    )
+    .map_err(|e| Errors::SshCertsError(e))?;
```

Comment:
Would call this SigningError. Even though we're using the `sshcerts` library, I don't think that context is important at this level